### DISCUSSION TITLE
RFID tag and detector

### DISCUSSION THREAD

**Discussion by ShuffleWire** [cyberbotics/webots#3857] [Q&A]
I'm working with RFID tag (specifically Active Reader Passive Tag (ARPT)), and I would like to be able to get a similar behaviour than the real one.
When a detector is near (distance and angle could be parameters), it should return (static?) data (for instance an ID) held by the tag.

I've thought about using Emitter (the RFID tag) and Receiver (the detector), but my tags are on a passive solid, and Emitters could be on robots only. I could have transform my Solid into Robot, but it's doesn't seems natural... Moreover I would have to define new controllers per Robot aka RFID tag and that don't seems to me to be the good way to go nor very handy. 

I'm currently using bumper to detect if my detector is near a tag, and then I guess the data (id of a object). (for the moment in my simulation I've only one object, but I will later have multiples object to detect).

**Comment by omichel** [comment]
I believe there are two options for implementing this in a clean way:
1. Create a new `RFIDReader` sensor in Webots (you can probably take inspiration from the `Receiver` sensor). The RFID sensor should be able to read the `description` field of the `Solid` nodes which have a `model` field set to `"RFID tag"`. 
2. As you suggested, implement your RFID tags as `Emitter` nodes, but set them all as children of a supervisor Robot that will emit permanently messages on all of them on a short range. This is however more "hacky" than the first option.

**Comment by ShuffleWire** [comment]
Well, as it turn out I will need to convert all my Solid object to Robot, so I could simply use a regular RFID and a small controller.
My use case is : I need to grab object on the floor using an arm, and read after that the id of the selected object. Now I've a bunch of Robot, executing the same controller, but using the "argv" data to send different data through this emitter.
